akka-remote/src/main/scala/akka/remote/artery/MessageDispatcher.scala

```diff
@@ -27,58 +28,66 @@ private[akka] class MessageDispatcher(
 
   private val remoteDaemon = provider.remoteDaemon
   private val log = Logging(system, getClass.getName)
+  private val debugLogEnabaled = log.isDebugEnabled
 
   def dispatch(
     recipient:        InternalActorRef,
     recipientAddress: Address,
     message:          AnyRef,
     senderOption:     OptionVal[ActorRef]): Unit = {
 
-    import provider.remoteSettings._
+    import provider.remoteSettings.Artery._
+    import Logging.messageClassName
 
     val sender: ActorRef = senderOption.getOrElse(system.deadLetters)
     val originalReceiver = recipient.path
 
-    def msgLog = s"RemoteMessage: [$message] to [$recipient]<+[$originalReceiver] from [$sender()]"
-
     recipient match {
 
       case `remoteDaemon` ⇒
-        if (UntrustedMode) log.debug("dropping daemon message in untrusted mode")
-        else {
-          if (LogReceive) log.debug("received daemon message {}", msgLog)
+        if (UntrustedMode) {
+          if (debugLogEnabaled) log.debug(
+            "dropping daemon message [{}] in untrusted mode",
+            messageClassName(message))
```

Comment:
We don't have that here, but we can probably pass it in from the stage that is calling this. Good idea.